### DISCUSSION TITLE
What routes must be proxied through to pocketbase?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3321] [Q&A] [answered]
I want to put pocketbase behind a proxy so static assets can be served from CDN rather than forwarding those requests to the pocketbase process (origin). Is there a regex or something I can use to forward requests to the origin vs safely intercept and respond with the static asset or 404?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure I follow.

If by static assets you mean files such as those in `pb_public`, then the base address to the files is `https://yourdomain.com/*` (_or the path of your custom static serve route_). You should be able to replace the url with your CDN address without extra forwarding.

If by static assets you mean the uploaded files in a collection (_aka. the files served via `/api/files/*`_), then it is a little more complicated because of 3 things:
- it depends on what storage implementation is used (local filesystem or S3; also note that most CDN services are eventually consistent and the file may not be immediately available after upload/cache - see https://github.com/pocketbase/pocketbase/issues/28)
- thumbs for image files are generated lazily on their first access, so you'll need some conditional logic that will check if a thumb file exists in the CDN or forward the request to the PocketBase server to generate it and then cache it with the CDN
- you'll need some custom handling if you want to serve [protected files](https://pocketbase.io/docs/files-handling/#protected-files) with a CDN (we use a short lived file token)

All uploaded collection files have the following path pattern: **`/:collectionId/:recordId/:filename`** (_generated thumbs are stored in an extra "thumbs_filename" subdirectory for easier deletion_).
When using the local filesystem, the files are stored in `pb_data/storage/:collectionId/:recordId/:filename`
When using S3, all files are stored in the configured bucket and the above path is used as a name prefix, aka. the full name of a single object is `/:collectionId/:recordId/:filename`.

**Comment by benallfree** [comment]
Interesting. My exact use case is optimizing the https://pockethost.io proxy to serve `pb_public` without spawning the underlying `pocketbase` executable to do it. I think anything matching `/^api\//` should probably be passed along to the executable. Do you agree?

edit: `/^api\//` and `/^_\//`

**Reply by ganigeorgiev** [reply]
For the default/system routes passing all `/api/*` requests to the executable should work for most of the times.

The only eventual issues I can think of are:
- custom user routes defined with `pb_hooks`
- you may need to enable on proxy level `index.html` forwarding on missing file for SPA applications that uses "pretty urls" (_I think this is enabled by default with the prebuilt exacutable; eg. `example.com/about/me` should actually serves `example.com/index.html` if `/about/me` file is not found so that the SPA router can handle the url path_)

**Reply by benallfree** [reply]
Yeah as we discuss it, it doesn't seem that important. The likelihood of a frontend needing only static assets and not a single backend request seems low. I'll postpone this idea for now.
